### PR TITLE
IOI Mountains Sparse Segtree Solution

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/platinum/ioi-05-mountain.mdx [hunk 1]
```diff
@@ -7,7 +7,121 @@ author: Justin Ji
...
+	}
+
+  public:
+	SparseSegtree(int n) : n(n) {}
+
+	inline void range_set(int ql, int qr, int val) {
+		range_set(root, 0, n - 1, ql, qr, val);
+	}
+
+	inline int walk_right(int mx) {
```
Comment (SansPapyrus683) [inline solutions/platinum/ioi-05-mountain.mdx:91]: ok this defo doesn't matter but idt `inline` does anything here
it is just a hint to the compiler so

**Review by SansPapyrus683 — APPROVED** [review]
tick tock
heavy like a brinks truck

File: solutions/platinum/ioi-05-mountain.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/platinum/ioi-05-mountain.mdx:91]: fair enough, i can remove it later

Comment (TheGamingMousse) [reply solutions/platinum/ioi-05-mountain.mdx]: ok removed it, vscode said that the compiler was inlining it anyways lmao

Comment (SansPapyrus683) [reply solutions/platinum/ioi-05-mountain.mdx]: yeah the compiler just does whatever the f it wants

**Review by ryanchou-dev — APPROVED** [review]
you cooked

**Comment by ryanchou-dev** [comment]
> tick tock heavy like a brinks truck

since when did kevin listen to joji 🤔

**Pull request merged**